src/Domain/CoordinateMaps/Frustum.cpp

```diff
@@ -0,0 +1,339 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/Frustum.hpp"
+
+#include "DataStructures/MakeWithValue.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/DereferenceWrapper.hpp"
+
+namespace CoordinateMaps {
+Frustum::Frustum(const std::array<std::array<double, 2>, 4> face_vertices,
+                 const double lower_bound, const double upper_bound,
+                 const Direction<3> direction_of_frustum) noexcept
+    : x1_(face_vertices[0][0]),
```

Comment:
instead of all these things, have two arrays of arrays and just have the constructor eat the inputs